libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2024 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/** Transformer to standardize DOI references and remove dx.doi.org */
+publish transform RemoveDxDotDoiDotOrg {
+	from valueWithBraces oftype text;
+	to valueWithoutBraces oftype text;
+
+	valueWithoutBraces: valueWithBraces replace /\b(dx\.doi\.org\/)\b/ with "";
+}
+
+/** Transformer to standardize DOI references and remove http:/**dx.doi.org/ */
+publish transform RemoveHttpDxDotDoiDotOrg {
+	from valueWithBraces oftype text;
+	to valueWithoutBraces oftype text;
+
+	valueWithoutBraces: valueWithBraces replace /^http:\/\/dx\.doi\.org\/\b/ with "";
+}
+
+/** Transformer to standardize DOI references and remove http:/** */
+publish transform RemoveHttp {
+	from valueWithBraces oftype text;
+	to valueWithoutBraces oftype text;
+
+	valueWithoutBraces: valueWithBraces replace /(\bhttp:\/\/)/ with "";
+}
+
+/** Transformer to standardize DOI references and remove https://doi.org/ */
+publish transform RemoveHttpsDoiDotOrg {
+	from valueWithBraces oftype text;
+	to valueWithoutBraces oftype text;
+
+	valueWithoutBraces: valueWithBraces replace /^https:\/\/doi\.org\/\b/ with "";
+}
+
+/** Forces all values in a column to follow the doi format. Removes prefixes like "http://" */
```

Comment:
Please extend the documentation a bit, it is really important for this to be good in the standard library. For the DOIStandardizer that would mean some inputs and how they would look standardized (e.g. http://doi.org/123 -> 123 etc). I hope you can look at the GTFS blocks in the transport extension for examples of docs.